File: manager/src/main/java/moe/shizuku/manager/utils/EnvironmentUtils.java
package moe.shizuku.manager.utils;

import android.app.UiModeManager;
import android.content.Context;
import android.content.res.Configuration;

public class EnvironmentUtils {

    private EnvironmentUtils() {
    }

    public static boolean isWatch(Context context) {
        return context.getSystemService(UiModeManager.class).getCurrentModeType()
                == Configuration.UI_MODE_TYPE_WATCH;
    }
}

File: manager/src/main/java/moe/shizuku/manager/utils/EnvironmentUtils.kt
package moe.shizuku.manager.utils

import android.app.UiModeManager
import android.content.Context
import android.content.res.Configuration
import android.os.SystemProperties
import java.io.File

object EnvironmentUtils {

    @JvmStatic
    fun isWatch(context: Context): Boolean {
        return (context.getSystemService(UiModeManager::class.java).currentModeType
                == Configuration.UI_MODE_TYPE_WATCH)
    }

    fun isRooted(): Boolean {
        return System.getenv("PATH")?.split(File.pathSeparatorChar)?.find { File("$it/su").exists() } != null
    }

    fun getAdbTcpPort(): Int {
        var port = SystemProperties.getInt("service.adb.tcp.port", -1)
        if (port == -1) port = SystemProperties.getInt("persist.adb.tcp.port", -1)
        return port
    }
}
